components/CallToActionContent.vue

```diff
@@ -1,20 +1,38 @@
 <template>
-  <section class="container mx-auto text-center py-6 mb-12">
-    <h1 class="w-full my-2 text-5xl font-bold leading-tight text-center text-white">Call to Action</h1>
-    <div class="w-full mb-4">
-      <div class="h-1 mx-auto bg-white w-1/6 opacity-25 my-0 py-0 rounded-t"></div>
-    </div>
+  <section class="container call-to-action-content">
+    <the-title :isLight="true" label="Call to Action" />
 
-    <h3 class="my-4 text-3xl leading-tight">Main Hero Message to sell yourself!</h3>
+    <h3>Main Hero Message to sell yourself!</h3>
 
-    <button
-      class="mx-auto lg:mx-0 hover:underline bg-white text-gray-800 font-bold rounded-full my-6 py-4 px-8 shadow-lg"
-    >Action!</button>
+    <button>Action!</button>
   </section>
 </template>
 
 <script>
+import TheTitle from '@/components/TheTitle'
+
 export default {
-  name: 'CallToActionContent'
+  name: 'CallToActionContent',
+  components: {
+    'the-title': TheTitle
+  }
 }
 </script>
+
+<style scoped>
+.call-to-action-content {
```

Comment:
can you explain me why you created a class to use the apply rule instead of using the four classes inside the template?

Reply:
I wanted to follow the concept separation of concerns (html for markup - css for styling) as strict as possible, so the template should contain as little styling information as possible without making it too complicated. At some points it may've been useful to add a class name instead of using the element tag as the selector (easier to access/overwrite from an upper scope and more flexibility in usage of tag).

Reply:
hm, but isn't that then component based CSS again? so instead of @apply utility CSS classes I could just have use the usual statements.

the advantage of utility css classes is that you don't think in components anymore and you don't have to think about names.

only if something is used a lot of times (like container or boxes) I would create a class for that.

what is the advantage of tailwindcss when we still would create classes?